gm_dotnet_managed/GmodNET/Lua.cs

```diff
@@ -367,11 +368,28 @@ public void PushCFunction(CFuncManagedDelegate managed_function, bool use_safe_e
             }
         }
 
-        public void PushCFunction(IntPtr native_func_ptr)
+        public unsafe void PushCFunction(IntPtr native_func_ptr)
         {
+            if(native_func_ptr == IntPtr.Zero)
+            {
+                throw new ArgumentNullException("native_func_ptr", "Parameter can't be nullptr.");
+            }
+
             push_c_function(ptr, native_func_ptr);
         }
 
+        public unsafe void PushCFunction(delegate* unmanaged[Cdecl]<IntPtr, int> function_pointer)
+        {
+            IntPtr int_ptr = (IntPtr)function_pointer;
+
+            if (int_ptr == IntPtr.Zero)
+            {
+                throw new ArgumentNullException("function_pointer", "Parameter can't be nullptr.");
+            }
+
+            push_c_function(ptr, (IntPtr)function_pointer);
```

Comment:
Use local var int_ptr instead of casting again

Reply:
Fixed